tests/check_ssid_wifi.py

```diff
@@ -36,41 +36,72 @@ def WifiTest(self):
             self.link_up()
             
             '''Generate WPA supplicant file and execute it'''
-            wlan.sendline("rm /etc/"+ssid_name+".conf")
-            wlan.expect(prompt)
-            wlan.sendline("wpa_passphrase "+ssid_name+ " >> /etc/"+ssid_name+".conf")
-            wlan.expect("")
-            wlan.sendline(pass_word)
-            wlan.expect(prompt)
-            wlan.sendline("cat /etc/"+ssid_name+".conf")
-            wlan.expect(prompt)
-            wlan.sendline("wpa_supplicant  -B -Dnl80211 -iwlan1 -c/etc/"+ssid_name+".conf")
-            wlan.expect(prompt)
+            self.wpa_connect(ssid_name,pass_word)
 
-            '''Ping test'''
-            self.ping_hostname(ssid_name)
+            '''Check wlan connectivity'''
+            self.wlan_connect(ssid_name)
+
+            '''Ping gateway test'''
+            self.ping_gateway()
+
+            '''Ping hostname test'''
+            self.ping_hostname()
```

Comment:
This looks better... but these functions should be on devices/debian_wifi.py not on the test class.